frontend/src/components/NextPhaseButton/NextPhaseButton.tsx

```diff
@@ -0,0 +1,40 @@
+import rootStore from '../../stores/RootStore';
+import Button from '../Button/Button';
+import { observer } from 'mobx-react';
+
+let buttonLabel = 'Next phase';
+
+export enum GamePhases {
+  TILE_PLACEMENT = 1,
+  MEEPLE_PLACEMENT,
+  SCORE_PHASE,
+}
+
+export const NextPhaseButton = observer(() => {
+  const currentPhase = rootStore.gameStore.currentPhase;
+  const numbers = [1, 2, 3];
+  const handleNextPhase = () => {
+    rootStore.gameStore.setNextPhase();
+    if (currentPhase === 2) {
+      buttonLabel = 'End turn';
+    } else {
+      buttonLabel = 'Next phase';
```

Comment:
But we shouldn't do this here (also this label as let outside the component is kinda weird)...

Reply:
Instead you can simply <Button text={ currentPhase ===  GamePhases.SCORE_PHASE ? "End turn" : "Next phase" }